src/sqlfluff/rules/L003.py

```diff
@@ -776,136 +660,358 @@ def _process_current_line(
+        # to the next, making it a bit unclear how to indent that line.
+        def _find_matching_start_line(
+            previous_lines: List[_LineSummary],
+        ) -> Optional[_LineSummary]:
+            template_block_level = -1
+            for template_line in previous_lines:
+                if not template_line.templated_line_type:
+                    continue
+                if template_line.templated_line_type == "end":
+                    template_block_level -= 1
                 else:
-                    break
+                    template_block_level += 1
 
-            if fixes:
-                return LintResult(
-                    anchor=anchor,
-                    memory=memory,
-                    description="Comment not aligned with following line.",
-                    fixes=fixes,
-                )
+                if template_block_level != 0:
+                    continue
+
+                return template_line
+            return None  # pragma: no cover
 
-            # Otherwise all good.
+        template_line = _find_matching_start_line(previous_lines)
+        # This Cant occur in valid code
+        assert template_line, "TypeGuard"
+
+        if template_line.line_no in memory.noncomparable_lines:
             return LintResult(memory=memory)
 
-            # NB: At shallower indents, we don't check, we just check the
-            # previous lines with the same balance. Deeper indents can check
-            # themselves.
+        self.logger.debug(
+            "    [template block end] Comparing to #%s", template_line.line_no
+        )
+        if this_line.indent_size == template_line.indent_size:
+            return LintResult(memory=memory)
 
-        # If we get to here, then we're all good for now.
-        return LintResult(memory=memory)
+        memory.problem_lines.add(this_line.line_no)
 
-    @classmethod
-    def _get_element_template_info(
-        cls, elem: BaseSegment, templated_file: Optional[TemplatedFile]
-    ) -> Optional[str]:
-        if elem.is_type("placeholder"):
-            if templated_file is None:
-                raise ValueError("Parameter templated_file cannot be: None.")
-            assert elem.pos_marker
-            slices = templated_file.raw_slices_spanning_source_slice(
-                elem.pos_marker.source_slice
-            )
-            if slices:
-                return slices[0].slice_type
-        return None
+        # The previous indent.
+        desired_indent = "".join(elem.raw for elem in template_line.indent_buffer)
+        first_non_indent_i = len(this_line.indent_buffer)
+        current_anchor = this_line.line_buffer[first_non_indent_i]
+        fixes = self._coerce_indent_to(
+            desired_indent=desired_indent,
+            current_indent_buffer=this_line.indent_buffer,
+            current_anchor=current_anchor,
+        )
+        self.logger.debug(
+            "    !! Indentation does not match #%s." " Fixes: %s",
```

Comment:
Can we combine the two strings on this line? I assume this was previously two lines that were combined.

Reply:
Which two string?

Reply:
Look carefully at the `"` on this line. I think there are 4, thus two different strings adjacent to one another.